File: BW5/BW-5/src/app/interfaces/cliente.ts
import { Fatture } from "./fatture"

export interface Cliente {

        id: number,
        dataInserimento: Date,
        dataUltimocontatto: Date,
        ragioneSociale: string,
        partitaIva: string,
        email: string,
        tipoCliente : string ,
        fatturatoAnnuale : number,
        pec : string ,
        telefono : number ,
        emailContatto :  string ,
        nomeContatto : string ,
        cognomeContatto : string ,
        telefonoContatto : number ,
        indirizzi : [
            {
                via : string ,
                comune : any ,
                cap :  number ,
                localita :  string ,
                civico : number,
                tipoIndirizzo : string
            }
        ]
    }

File: BW5/gruppo-3/src/main/java/BW_5/gruppo_3/entity/Cliente.java
package BW_5.gruppo_3.entity;

import BW_5.gruppo_3.Enums.TipoCliente;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDate;
import java.util.List;

@Entity
@Data
public class Cliente {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String ragioneSociale;

    private String partitaIva;

    private String email;

    @Enumerated(EnumType.STRING)
    private TipoCliente tipoCliente;

    private LocalDate dataInserimento;

    private LocalDate dataUltimoContatto;

    private double fatturatoAnnuale;

    private String pec;

    private String telefono;

    private String emailContatto;

    private String nomeContatto;

    private String cognomeContatto;

    private String telefonoContatto;

    private String logoAziendale;

    @OneToMany(mappedBy = "cliente")
    private List<Indirizzo> indirizzi;


    @OneToMany(mappedBy = "cliente")
    @JsonIgnore
    private List<Fattura> fatture;


}
